net/Socket.hpp

```diff
@@ -1153,17 +1152,18 @@ class StreamSocket : public Socket,
             ssize_t len;
             do
             {
-                // Writing more than we can absorb in the kernel causes SSL wastage.
-                len = writeData(&_outBuffer[0], std::min((int)_outBuffer.size(),
-                                                         getSendBufferSize()));
+                // Writing much more than we can absorb in the kernel causes wastage.
+                len = writeData(_outBuffer.getBlock(),
+                                std::min((int)_outBuffer.getBlockSize(),
```

Comment:
Any reason not to use `_outBuffer.size()` here?

Reply:
Sure that's easy - we'll switch to having blocks/chunks under the hood that are not all in the same allocations in some star-trek future where we have time =)